libraries/apollo-runtime/src/commonMain/kotlin/com/apollographql/apollo3/network/ws/WebSocketNetworkTransport.kt

```diff
@@ -353,11 +354,24 @@ private constructor(
     private var idleTimeoutMillis: Long? = null
     private var protocolFactory: WsProtocol.Factory? = null
     private var reopenWhen: (suspend (Throwable, attempt: Long) -> Boolean)? = null
+    private var reopenServerUrl: (suspend () -> String)? = null
 
     fun serverUrl(serverUrl: String) = apply {
       this.serverUrl = serverUrl
     }
 
+    /**
+     * Configure dynamically the server URL.
+     *
+     * @param serverUrl a function return the new server URL to use when reopening the [WebSocketNetworkTransport] containing
+     * the new auth credentials for example in case of an unauthorized error with the web socket.
```

Comment:
```suggestion
     * @param serverUrl a function returning the new server URL.
     * This function will be called every time a WebSocket is opened. For example, you can use it to update your
     * auth credentials in case of an unauthorized error.
```